plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1155,50 +1159,90 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         catalog.renameTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName(), newTable);
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
avoid suppression on a large method; if a suppression is needed, make it variable scoped (assignment)